api/system/services/System.js

```diff
@@ -1,8 +1,106 @@
 'use strict';
+const fetch = require('node-fetch');
+
 
 /**
  * Read the documentation (https://strapi.io/documentation/3.0.0-beta.x/guides/services.html#core-services)
  * to customize this service
  */
 
-module.exports = {};
+module.exports = {
+  /**
+   * Promise to add record
+   *
+   * @return {Promise}
+   */
+
+  create: async (values) => {
+    let edsmData = await strapi.services.system.edsmUpdate(values);
+
+    return strapi.query('System').create(edsmData);
+  },
+
+  /**
+   * Promise to edit record
+   *
+   * @return {Promise}
+   */
+
+  update: async (params, values) => {
+
+    if (!values.edsmCoordLocked){
+      let oldData = await strapi.services.system.findOne({id: params.id});
+      oldData = oldData.toJSON();
+
+      if (oldData.edsmCoordLocked == false || oldData.edsmCoordLocked == undefined) {
```

Comment:
If you're checking against boolean false you should consider using `===` instead of `==`. `==` will convert types for you so `edsmCoordLocked: 0` will be false.

Same goes for `undefined`, you're checking a variable, so I can do

```
const undefined = 'something';
if( oldData.edsmCoordLocked == undefined ) {
    // This will only be true when oldData.edsmCoordLocked is equal to 'something'
}
```

You can check for `undefined` like here:
```
if( typeof oldData.edsmCoordLocked === 'undefined' ) { ... }
```

`typeof` will return a string of the variable type ('object', 'function', 'boolean', 'null', 'undefined', etc...)